components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,336 @@
+        when(secretManager.getSecretType(any())).thenReturn(secretType);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        identityTenantUtil.close();
+        identityDatabaseUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+
+        closeH2Database();
+    }
+
+    @Test(priority = 1)
+    public void testAddAction() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(false);
+        action = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, creatingAction.getId(), creatingAction, TENANT_ID);
+        Assert.assertEquals(creatingAction.getId(), action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertEquals(creatingAction.getDescription(), action.getDescription());
+        Assert.assertEquals(creatingAction.getType(), action.getType());
+        Assert.assertEquals(creatingAction.getStatus(), action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(), action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 2)
+    public void testGetActionsByActionType() throws ActionMgtException, SQLException {
+
+        Assert.assertEquals(1, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+        mockDBConnection();
+        Action result = daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).get(0);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 3)
+    public void testDeleteAction() throws ActionMgtException, SQLException {
+
+        daoImpl.deleteAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), action, TENANT_ID);
+        mockDBConnection();
+        Assert.assertEquals(0, daoImpl.getActionsByActionType(PRE_ISSUE_ACCESS_TOKEN, TENANT_ID).size());
+    }
+
+    @Test(priority = 4)
+    public void testAddActionWithoutOptionalAttributes() throws ActionMgtException {
+
+        Action creatingAction = buildMockAction(true);
+        action = daoImpl.addAction(PRE_ISSUE_ACCESS_TOKEN, creatingAction.getId(), creatingAction, TENANT_ID);
+        Assert.assertEquals(creatingAction.getId(), action.getId());
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertNull(null,action.getDescription());
```

Comment:
There's a minor formatting problem here just after ','

Reply:
Fixed in commit 8bb085dbb1b8c3661a73f59eeb300674c7bc0dfd